### PR TITLE
Using sidecar handler to run Papermill task

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
https://flyte-org.slack.com/archives/CP2HDHKE1/p1661288462596299
If we use `pod_config` in Notetask, the task type will become `nb-sidecar`, which is incorrect. 
```bash
{"json":{"exec_id":"ahkm5jd76h8l2gn46zt6","node":"n1","ns":"flyteexamples-development","res_ver":"209788243","routine":"worker-2","tasktype":"nb-
sidecar","wf":"flyteexamples:development:flyte.workflows.simple.nb_to_python_wf"},"level":"warning","msg":"No plugin 
found for Handler-type [nb-sidecar], defaulting to [container]","ts":"2022-edited"}

```
To trigger the [sidecar](https://github.com/flyteorg/flyteplugins/blob/c4fe3118a8a4f7d5f77288c6bd37c2b32743dbb2/go/tasks/plugins/k8s/pod/sidecar.go#L18) handler, 
- We should remove the prefix `nb-`.
- Add pod spec to task template by overriding [get_k8s_pod](https://github.com/flyteorg/flytekit/blob/571fc89013ad9506d283a9ed3d0a88de6004a064/flytekit/tools/translator.py#L175) and `get_container`. When using pod config in notebook task, the container spec will become None

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

File: plugins/flytekit-papermill/flytekitplugins/papermill/task.py [hunk 1]
```diff
@@ -10,10 +10,12 @@
 from nbconvert import HTMLExporter
 
 from flytekit import FlyteContext, PythonInstanceTask
+from flytekit.configuration import SerializationSettings
 from flytekit.core.context_manager import ExecutionParameters
 from flytekit.deck.deck import Deck
 from flytekit.extend import Interface, TaskPlugins, TypeEngine
 from flytekit.loggers import logger
+from flytekit.models import task as _task_models
```
Comment (wild-endeavor) [inline plugins/flytekit-papermill/flytekitplugins/papermill/task.py:18]: can we not use the `_` anymore?  let's just say `task_models`

**Review by wild-endeavor — DISMISSED** [review]

File: plugins/flytekit-papermill/flytekitplugins/papermill/task.py (hunk 1 above)
Comment (pingsutw) [reply plugins/flytekit-papermill/flytekitplugins/papermill/task.py]: Updated it.

**Review by wild-endeavor — APPROVED** [review]
